src/lightning_lite/__init__.py

```diff
@@ -13,6 +14,12 @@
     _logger.propagate = False
 
 
+# In PyTorch 1.14+, setting this variable will force `torch.cuda.is_available()` and `torch.cuda.device_count()`
+# to use an NVML-based implementation that doesn't poison forks.
+# https://github.com/pytorch/pytorch/issues/83973
+os.environ["PYTORCH_NVML_BASED_CUDA_CHECK"] = "1"
```

Comment:
I would suggest moving this into the `cuda.py` file. It does get imported by default anyways.

I guess you chose to set it on import in case users make calls before the Trainer starts, additionally, I would also set it on setup and pop it on teardown.

Reply:
I want this to happen before anything else runs in Lightning, so that we don't shoot ourselves in the foot in the future when we import something that checks `torch.cuda.*`. 

This is an env variable for PyTorch, we don't need to tear it down. And we also don't wan to tear it down between Trainer calls for example, because if you poison a fork after trainer.fit(), you won't be able to run trainer.test() afterwards (ddp_fork in a notebook)